feat(service-worker): Improves storage full detection in data caching

SkyZeroZx opened the pull request:
This commit improves storage full detection capabilities in Angular Service Workers.

The change includes:
- Added `detectStorageFull()` method to DataGroup class for proactive storage monitoring
- Implemented Storage API integration to check usage and quota information
- Replaced TODO comments with actual storage capacity monitoring
- Added comprehensive error handling for Storage API calls
- Integrated storage detection into existing error handling workflows

### Use Cases
The storage quota detection feature is particularly useful for:
**Proactive storage management**: Detecting when browser storage is approaching capacity before cache operations fail

### Proposed Solution
- Add `detectStorageFull()` private method that uses `navigator.storage.estimate()` API
- Implement 95% threshold detection to provide safety buffer before actual storage exhaustion

SkyZeroZx committed:
feat(service-worker): Improves storage full detection in data caching

Adds proactive storage capacity checks using the Storage API
to better handle near-full conditions during data caching.
Replaces previous TODOs with actual usage monitoring and
logging to help prevent cache failures due to quota exhaustion.

atscott commented on packages/service-worker/worker/src/data.ts:
```diff
@@ -624,4 +622,38 @@ export class DataGroup {
+      }
+    } catch (err) {
+      // Storage API call failed
+      this.debugHandler.log(
```
If the storage API is throwing, I can imagine it will always throw. Might it be better to just ignore the error or at least ensure this isn't logged more than once?

atscott replied:
I think ignoring the error is reasonable due to baseline support. This is also only a debug/helpful check, not something that's required for things to work so the error doesn't _need_ to be reported

SkyZeroZx replied:
@atscott  updated it

SkyZeroZx committed:
fixup! feat(service-worker): Improves storage full detection in data caching

Adds proactive storage capacity checks using the Storage API
to better handle near-full conditions during data caching.
Replaces previous TODOs with actual usage monitoring and
logging to help prevent cache failures due to quota exhaustion.

thePunderWoman commented:
This PR was merged into the repository by commit b65c3d5e195267dd90b2826d4615ced1328b1709.

The changes were merged into the following branches: main
